pallets/schemas/src/lib.rs

```diff
@@ -198,21 +198,26 @@ pub mod pallet {
 		/// * [`Error::InvalidSchema`] - Schema is malformed in some way
 		/// * [`Error::SchemaCountOverflow`] - The schema count has exceeded its bounds
 		///
+		// Wondering how the total number of Schemas affects in this case to calculate the weight
+		// Also why 1_000 is used. Shouldn't be MaxSchemaRegistrations as worst case?
 		#[pallet::weight(< T as Config >::WeightInfo::create_schema(model.len() as u32, 1000))]
 		pub fn create_schema(
 			origin: OriginFor<T>,
 			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
 			model_type: ModelType,
 			payload_location: PayloadLocation,
 		) -> DispatchResult {
+			// Anyone with enough balance to pay fot the tx can create Schemas?
+			// Considering there is a fixed max number of those, might be a good idea limiting access to this method.
+			// You could for example lock/resrve some tokens (deposit for creating the Schame) or add a especial origin (Governance and Providers?)
 			let sender = ensure_signed(origin)?;
 
 			ensure!(
 				model.len() > T::MinSchemaModelSizeBytes::get() as usize,
 				Error::<T>::LessThanMinSchemaModelBytes
 			);
 			ensure!(
-				model.len() < Self::get_schema_model_max_bytes() as usize,
+				model.len() < Self::get_schema_model_max_bytes() as usize, // < or <= ?
```

Comment:
Should likely be inclusive instead of exclusive. @wilwade to add an issue for it.

Reply:
https://github.com/LibertyDSNP/frequency/issues/755